File: example-logs/logs.py
#!/usr/bin/env python

logfile = "/user/hdfs/data/weblogs/weblogs/2013-09-15.log"
log = sc.textFile(logfile)

jpglog = log.filter(lambda x: ".jpg" in x)
jpglog.take(10)
jpglog.count()

log.map(lambda s: len(s)).take(5)

log.map(lambda s: s.split()).take(5)

ip = log.map(lambda s: s.split()[0])
ip.take(5)

for x in ip.take(5) : print x

ip.saveAsTextFile("/user/hdfs/python/iplist")

# All logs

logfiles = "/user/hdfs/data/weblogs/weblogs/"
logs = sc.textFile(logfiles)
htmllogs=logs.filter(lambda s: ".htm" in s).map(lambda s: (s.split()[0],s.split()[2]))
for x in htmllogs.take(10): print x[0]+"/"+x.[1]

File: example-logs/logs.scala
val mydata = sc.textFile("user/hdfs/ulabs/data/frostroad.txt")
mydata.count()
mydata.collect()

val logfile = "/user/hdfs/data/weblogs/weblogs/2013-09-15.log"
val log = sc.textFile(logfile)

val  jpglog = log.filter(line => line.contains(".jpg"))
jpglog.take(10)

jpglog.count()

log.map(_.length).take(5)

log.map(_.split(' ')).take(5)

val ip = log.map(_.split(' ')(0))

ip.take(5)

ip.take(5).foreach(println)

ip.saveAsTextFile("/user/hdfs/scala/iplist")

// Loading all weblogs IP / UserID

val logfiles = "/user/hdfs/data/weblogs/weblogs/"
val logs = sc.textFile(logfiles)
val htmllogs = logs.filter(_.contains(".html")).map(line => (line.split(' ')(0), line.split(' ')(2)))

htmllogs.take(10).foreach(t => println(t._1 + "/" + t._2))

